### PR TITLE
Hook on filterProductContent for adding ProductComments data

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Hook on filterProductContent for adding ProductComments data
| Type?         | improvement
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes PrestaShop/Prestashop#25372
| How to test?  | Cf. #25372

### CODE REVIEW

File: productcomments.php [hunk 1]
```diff
@@ -896,6 +896,30 @@ public function hookDisplayFooterProduct($params)
...
+        }
+        /** @var ProductCommentRepository $productCommentRepository */
+        $productCommentRepository = $this->context->controller->getContainer()->get('product_comment_repository');
+
+        $averageRating = $productCommentRepository->getAverageGrade($params['object']->id, (bool) Configuration::get('PRODUCT_COMMENTS_MODERATE'));
+        $commentsNb = $productCommentRepository->getCommentsNumber($params['object']->id, (bool) Configuration::get('PRODUCT_COMMENTS_MODERATE'));
+
+        $params['object']->productComments = [
+            'averageRating' => $averageRating,
+            'numComments' => $commentsNb,
```
Comment (kpodemski) [inline productcomments.php:919]:
```suggestion
            'nbComments' => $commentsNb,
```

Comment (Progi1984) [reply productcomments.php:919]: why ?

Comment (kpodemski) [reply productcomments.php:920]: nb stands for number, right?

so, it's:

`commentsNb` or `nbComments` :D

Comment (Progi1984) [reply productcomments.php]: Fixed

**Review by kpodemski — APPROVED** [review]

**Comment by kpodemski** [comment]
I think we need to create a new milestone and prepare PRs for v4.2.3

**Comment by matks** [comment]
> I think we need to create a new milestone and prepare PRs for v4.2.3

If you feel like it's important, you can create it 😉  and modify PR milestones.

Just make sure changes are correctly targeting 5.0.0 or 4.2.3 . If we created milestone 5.0.0 I think it's because BC breaks have been introduced.

**Comment by Progi1984** [comment]
@matks & @kpodemski We need to realease `productcomments` to PS 1.7.8.1. Because there is a fix for this patch version.

**Comment by Progi1984** [comment]
In this I will rename the upgrade file.

**Comment by Progi1984** [comment]
@matks & @kpodemski I moved the upgrade file from 4.2.3 to 5.0.0. If you have any time, please review.

**Review by kpodemski — APPROVED** [review]

File: productcomments.php [hunk 2]
```diff
@@ -896,6 +896,31 @@ public function hookDisplayFooterProduct($params)
...
+    /**
+     * Inject data about productcomments in the product object for frontoffice
+     *
+     * @param array $params
+     *
+     * @return void
+     */
+    public function hookFilterProductContent(array $params)
+    {
+        if (empty($params['object']) || empty($params['object']->id)) {
```
Comment (PierreRambaud) [inline productcomments.php:908]:
```suggestion
        if (empty($params['object']->id)) {
```

Should be enough afaik

Comment (Progi1984) [reply productcomments.php]: Fixed

**Review by matthieu-rolland — APPROVED** [review]

**Comment by florine2623** [comment]
Hello @Progi1984 ,

Tested all features of module productcomments, all good to me !

Thanks !

**Comment by Progi1984** [comment]
Thanks @florine2623

**Pull request merged**